public/home.js

```diff
@@ -1,6 +1,216 @@
 $(document).ready(function(){
 
+  var APIaction = function(){
+    
+  };
 
+  APIaction.prototype.logOut = function(){
+    $.ajax({
+      type: 'DELETE',
+      url: '/sessions',
+      success: function(response){
+        alert("Logout successful. See you soon!")
+        window.location.href = '/';
+      }
+    })
+  };
+
+  APIaction.prototype.changeDetail = function(){
+    $.ajax({
+      type: 'PUT',
+      url: '/users',
+      data:{
+        user: {
+          firstname: firstName,
+          lastname: lastName,
+          email: email,
+          username: userName,
+          password: password,
+          dateCreated: new Date()
+        }
+      },
+      dataType: 'json',
+      success: function(response){
+        alert("Update successful!")
+        window.location.href = '/home';
+      },
+      error: function(xhr, status, data){
+        console.log(xhr);
+      }
+    })
+  };
+
+  APIaction.prototype.nameDisplay = function(){
+    $.ajax({
+      type: 'GET',
+      url: '/users',
+      dataType: 'json',
+      success: function(response){
+        html  =  '<h4>Welcome Back!<h4>'
+        html +=  '<h4>'+response.firstname.toUpperCase()+" "+response.lastname.toUpperCase()+'</h4>'
+        $('#name_display').replaceWith(html);
+      }
+    })
+  };
+
+  //wait up, building front end
+  APIaction.prototype.creatWorkOut = function(){
+    $.ajax({
+      type: 'POST',
+      url: '/workouts',
+      data:{
+        workout:{
+          title: title,
+          type: type,
+          startTime: startTime,
+          date: date,
+          duration: duration,
+          meetingLocation: meetingLocation,
+          comments: comments,
+        }
+      },
+      dataType: 'json',
+      success: function(response){
+        alert("Thank you for the post!")
+      },
+      error: function(xhr, status, data){
+        console.log(xhr);
+      }
+    })
+  }
+
+  var picChoice = function(type){
+    if(type == 'SWIM'){
+      return '/public/assets/swim.jpeg'
+    } else if (type == 'BIKE'){
+      return '/public/assets/bike.jpg'
+    } else if (type == 'RUN'){
+      return '/public/assets/run.jpeg'
+    } else if (type == 'TEAM'){
+      return '/public/assets/team.jpeg'
+    } else if (type == 'GROUP'){
+      return '/public/assets/bootcamp.jpeg'
+    } else if (type == 'OTHER') {
+      return '/public/assets/others.jpg'
+    }
+    
+  }
+
+
+  APIaction.prototype.getPostSuccess = function(response){
```

Comment:
Here is a perfect example of what I meant before about separating the API. In this case, you are mixing functions in APIaction that deal with `$.ajax` calls (API layer), with functions that interact with the html. Great job with the indentation btw :)